app/src/angular/classes/token.js

```diff
@@ -107,7 +107,10 @@ class Token {
 
             if(balanceHex !== oldBalanceHex){
                 $rootScope.$broadcast('balance:change', this.symbol, this.getBalanceDecimal(), this.balanceInUsd);
-                // TODO - scan transactions
+                
+                setTimeout(()=> {
+                    Web3Service.syncWalletActivityByContract();
```

Comment:
use **$timeout** instead of **setTimeout** or much better to listen **balance:change** event